src/docs/asciidoc/jme3/advanced/ragdoll.adoc

```diff
@@ -175,3 +173,87 @@ Read the <<jme3/advanced/physics#responding_to_a_physicscollisionevent,Respondin
+
+*  https://github.com/jMonkeyEngine/jmonkeyengine/blob/v3.2/jme3-examples/src/main/java/jme3test/bullet/TestBoneRagdoll.java[v3.2-branch TestBoneRagdoll.java]  Press the left mouse button to shoot a cannonball at the model.  Ragdoll physics starts when a cannonball touches the model.  Press the spacebar to resume kinematic mode and cause the model to rise to his feet. 
+
+
+== Ragdoll Physics using DynamicAnimControl (JME 3.3 and later)
+
+DynamicAnimControl is a replacement for KinematicRagdollControl.  The intent is similar, except that a DynamicAnimControl can be simultaneously kinematic for some bones and dynamic for others.  Also, DynamicAnimControl can be configured to work with a wider variety of models than KinematicRagdollControl.
+
+To simplify the implementation, the ragdoll created by a DynamicAnimControl is composed of "links".  Just as a RigidBodyControl connects a rigid body to a spatial, a link connects a rigid body to one or more bones in the model's skeleton.  Just like a RigidBodyControl, a link can be in kinematic mode or dynamic mode.
+
+And just as the bones in a skeleton are arranged in parent/child hierarchy, so are the links in a DynamicAnimControl ragdoll.  In the ragdoll hierarchy, there is exactly one link that has no parent; it is referred to as the "torso".  Every other link is a "bone link" which has another link as its parent.  Each bone link is connected to its parent by a physics joint.
+
+As with KinematicRagdollControl, you start by creating a control,
+
+[source,java]
+----
+ragdoll = new DynamicAnimControl();
+----
+
+but before adding it to the model, you must configure it by specifying the mass of the torso and also the mass and range of motion for each linked bone:
+
+[source,java]
+----
+ragdoll.setMass(DacConfiguration.torsoName, 1f);
+ragdoll.link("Waist", 1f, new RangeOfMotion(1f, -0.4f, 0.8f, -0.8f, 0.4f, -0.4f));
+...
+----
+
+You probably don't want to link every bone in the model's skeleton.  For instance, if the model has articulated fingers, you probably want to link the hand bones but not the individual finger bones.  Unlinked bones will be managed by their nearest linked ancestor, and the torso will manage any bones for which no ancestor is linked.  If you link too many bones, the ragdoll may become inflexible or jittery due to collisions between rigid bodies that don't share a physics joint.
+
+Only after the control is configured should you add it to the model and to physics space:
+
+[source,java]
+----
+model.addControl(ragdoll);
+getPhysicsSpace().add(ragdoll);
+----
+
+Note that the control must be added to the Spatial with the ArmatureControl, which isn't necessarily the model's root spatial.
```

Comment:
> ArmatureControl

@stephengold you mean `SkinningControl` ?

Reply:
Right.

Reply:
Thank you!